spec/views/images/crop.html.erb_spec.rb

```diff
@@ -0,0 +1,38 @@
+RSpec.describe "images/crop.html.erb" do
+  describe "image crop" do
+    it "shows a 'Save' button by with JS" do
+      assign(:image_revision, create(:image_revision))
+      assign(:edition, build(:edition))
+      render
+      expect(rendered).to have_button("Save")
+    end
+
+    it "shows a 'Continue' button without JS" do
+      assign(:image_revision, create(:image_revision))
+      assign(:edition, build(:edition))
+      render
+      expect(rendered).to have_selector('.app-no-js button[text()="Continue"]')
+    end
+
+    context "when the image has exact dimensions" do
+      it "shows a 'Continue' button instead" do
+        assign(:image_revision, create(:image_revision,
+                                       width: Image::WIDTH,
+                                       height: Image::HEIGHT))
+
+        assign(:edition, build(:edition))
+        render
+        expect(rendered).to have_selector('.app-js-only button[text()="Continue"]')
+      end
+    end
+
+    context "when uploading a new image" do
+      it "shows a 'Save and continue' button" do
+        assign(:image_revision, create(:image_revision))
+        assign(:edition, build(:edition))
+        render(template: "images/crop", locals: { params: { wizard: "upload" } })
```

Comment:
I'm not quite sure what the `template: "images/crop"` bit is doing. I'd have expected something like this instead, consistently with the others:

```
assign(:params, wizard: "upload")
```

Reply:
So the `template` part is necessary in order to override the default agruments of `render` (otherwise you'll get an error, so I'm going to say it's self explaining). We can't use `assign` here, as otherwise we'd need to be using `@params` in the view.

Reply:
Makes sense. Thanks for explaining!